corehq/apps/geospatial/static/geospatial/js/models.js

```diff
@@ -811,12 +818,197 @@ hqDefine('geospatial/js/models', [
         }
     };
 
+    var AssignmentRow = function (caseName, caseId, userId, userName, userPrimaryLocName, mapItem) {
+        let self = this;
+
+        self.caseName = ko.observable(caseName);
+        self.caseId = caseId;
+        self.assignedUsername = ko.observable(userName);
+        self.assignedUserId = userId;
+        self.assignedUserPrimaryLocName = ko.observable(userPrimaryLocName);
+        self.mapItem = mapItem;
+        self.isSelected = ko.observable(false);
+
+        self.getJson = function () {
+            return {
+                caseName: self.caseName(),
+                caseId: self.caseId,
+                assignedUsername: self.assignedUsername(),
+                assignedUserId: self.assignedUserId,
+                assignedUserPrimaryLocName: self.assignedUserPrimaryLocName(),
+            };
+        };
+
+        return self;
+    };
+
+    var AssignmentManager = function (mapModel, disbursementModel) {
+        let self = this;
+
+        self.itemsPerPage = ko.observable(5);
+        self.totalItems = ko.observable(0);
+        self.currentPage = ko.observable(1);
+
+        self.mapModel = mapModel;
+        self.disbursementModel = disbursementModel;
+
+        self.caseData = [];
+        self.filteredCaseData = ko.observableArray([]);
+        self.caseDataPage = ko.computed(function () {
+            const end = self.currentPage() * self.itemsPerPage();
+            const start = end - self.itemsPerPage();
+            return self.filteredCaseData().slice(start, end);
+        });
+
+        self.userData = ko.observableArray([]);
+        self.canOpenModal = ko.computed(function () {
+            return self.mapModel.caseGroupsIndex.length;
+        });
+
+        self.selectedUserId = ko.observable();
+        self.includeRelatedCases = ko.observable(false);
+
+        self.assignedFilter = ko.observable();
+        self.assignedFilter.subscribe(() => {
+            if (self.assignedFilter() === 'all') {
+                self.filteredCaseData(self.caseData);
+            } else {
+                self.filteredCaseData(self.caseData.filter(function (caseItem) {
+                    if (self.assignedFilter() === 'unassigned') {
```

Comment:
nitpicking here,  maybe consider moving this check outside of the loop ?
Would reduces the no. of checks however duplicates a bit of code.

Reply:
This could be refactored by creating the filter first with a conditional and then applying it. This would look like this:

```
 let assignedFilter;
  if (self.assignedFilter() === 'unassigned') {
      assignedFilter = (caseItem) => { return !caseItem.assignedUserId; };
  } else {
      assignedFilter = (caseItem) => { return caseItem.assignedUserId; };
  }
  self.filteredCaseData(self.caseData.filter((caseItem) => assignedFilter(caseItem)));
```

Is this what you had in mind? It does make the code a bit more bloated, and I'm not sure if this extra code complexity justifies the negligible performance gains from removing the conditional in the loop.

Reply:
I wad thinking of something simple below.

```
if (self.assignedFilter() === 'all') {
    self.filteredCaseData(self.caseData);
} else if (self.assignedFilter() === 'unassigned') {
    self.filteredCaseData(self.caseData.filter(function (caseItem) {
        return !caseItem.assignedUserId;
    }));
} else {
    self.filteredCaseData(self.caseData.filter(function (caseItem) {
        return caseItem.assignedUserId;
    }));
}
```

I am also good with the original approach as performance gains is close to none.

Reply:
I think given the negligible performance gains, I'll keep the implementation as it is. I feel like changing the code to one of the above suggestions makes the code a bit less DRY and adds additional complexity for minimal gains.